community/kernel/src/main/java/org/neo4j/kernel/api/index/util/FailureStorage.java

```diff
@@ -123,6 +124,18 @@ public synchronized void storeIndexFailure( long indexId, String failure ) throw
             channel.force( true );
             channel.close();
         }
+        catch ( FileNotFoundException fnf )
```

Comment:
Would it be more concise if there was a method that checks for and creates if absent file file which can cause this FNFE? That can be called before each operation, avoiding handling the exception and making clear what file is missing.
A step further could be to check that during construction, so all methods operate after the file has been verified to be in place, and fail (as they should) if the file is for some reason not found.